model/jpa/src/main/java/org/keycloak/events/jpa/JpaEventStoreProviderFactory.java

```diff
@@ -34,16 +33,18 @@ public class JpaEventStoreProviderFactory implements EventStoreProviderFactory,
 
     public static final String ID = "jpa";
     private int maxDetailLength;
+    private int maxFieldLength;
 
     @Override
     public EventStoreProvider create(KeycloakSession session) {
         JpaConnectionProvider connection = session.getProvider(JpaConnectionProvider.class);
-        return new JpaEventStoreProvider(session, connection.getEntityManager(), maxDetailLength);
+        return new JpaEventStoreProvider(session, connection.getEntityManager(), maxDetailLength, maxFieldLength);
     }
 
     @Override
     public void init(Config.Scope config) {
-        maxDetailLength = config.getInt("max-detail-length", 0);
+        maxDetailLength = config.getInt("max-detail-length", -1);
+        maxFieldLength = config.getInt("max-field-length", -1);
```

Comment:
These should be checked to be greater than 3 (otherwise they can lead to an exception later)